src/main/java/io/jenkins/stapler/idea/jelly/symbols/SymbolFinder.java

```diff
@@ -0,0 +1,26 @@
+package io.jenkins.stapler.idea.jelly.symbols;
+
+import com.intellij.openapi.project.Project;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public final class SymbolFinder {
+
+    private static final ConcurrentMap<String, Set<Symbol>> ICONS_CACHE = new ConcurrentHashMap<>();
+
+    private static final List<SymbolLookup> SYMBOL_LOOKUPS =
+            List.of(new LocalSymbolLookup(), new JenkinsSymbolLookup(), new IoniconsApiSymbolLookup());
```

Comment:
if you add a new symbol it reports it as invalid which isn't great.

Do we need to cache the local ones?

<img width="1161" alt="image" src="https://github.com/user-attachments/assets/b1c9f6cc-9f3f-4697-8ee2-b2128bf05676" />

Reply:
Or possibly invalidate the cache / index when a new SVG file is created? or something along those lines?

Reply:
Good idea, added a file listener that will trigger the cache to be cleared when an SVG is modified.

Reply:
Retested, looks good